packages/eslint-plugin/src/rules/padding-line-between-statements.ts

```diff
@@ -0,0 +1,783 @@
+import {
+  AST_NODE_TYPES,
+  TSESLint,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+import * as util from '../util';
+
+/**
+ * This rule is a replica of padding-line-between-statements.
+ *
+ * Ideally we would want to extend the rule support typescript specific support.
+ * But since not all the state is exposed by the eslint and eslint has frozen stylistic rules,
+ * (see - https://eslint.org/blog/2020/05/changes-to-rules-policies for details.)
+ * we are forced to re-implement the rule here.
+ *
+ * We have tried to keep the implementation as close as possible to the eslint implementation, to make
+ * patching easier for future contributors.
+ *
+ * Reference rule - https://github.com/eslint/eslint/blob/master/lib/rules/padding-line-between-statements.js
+ */
+
+//------------------------------------------------------------------------------
+// Local Types
+//------------------------------------------------------------------------------
+
+type NodeTest = (
+  node: TSESTree.Node,
+  sourceCode: TSESLint.SourceCode,
+) => boolean;
+
+interface NodeTestObject {
+  test: NodeTest;
+}
+
+interface PaddingOption {
+  blankLine: keyof typeof PaddingTypes;
+  prev: string | string[];
+  next: string | string[];
+}
+
+type MessageIds = 'expectedBlankLine' | 'unexpectedBlankLine';
+type Options = PaddingOption[];
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+const LT = `[${Array.from(
+  new Set(['\r\n', '\r', '\n', '\u2028', '\u2029']),
+).join('')}]`;
+const PADDING_LINE_SEQUENCE = new RegExp(
+  String.raw`^(\s*?${LT})\s*${LT}(\s*;?)$`,
+  'u',
+);
+
+/**
+ * Creates tester which check if a node starts with specific keyword.
+ * @param {string} keyword The keyword to test.
```

Comment:
could you please delete all of the JSDoc type annotations?

They're just duplicated TS types and it just adds a potential place that we can forget to update and fall out of sync.

```suggestion
 * @param keyword The keyword to test.
```